packages/android_intent/android/src/main/java/io/flutter/plugins/androidintent/IntentSender.java

```diff
@@ -158,7 +185,46 @@ Intent buildIntent(
         intent.setPackage(null);
       }
     }
+    Boolean isPackageFilterRequired = ignoredPackages != null && ignoredPackages.size() > 0;
+    if (isPackageFilterRequired) {
+      String packageToLaunch = choosePackageToLaunch(intent, ignoredPackages);
+      intent.setPackage(packageToLaunch);
+    }
 
     return intent;
   }
+
+  String choosePackageToLaunch(Intent intent, List<String> ignoredPackages) {
```

Comment:
This and the below helper methods should be `private`, and I think also `static`?

Reply:
I made them only private, but not static, because they use the instance fields `activity` and `applicationContext`, which I would have to pass through several static methods.